src/askvm/resources/core/let.ts

```diff
@@ -16,6 +16,21 @@ export const letRes = resource({
       throw new Error(`Key "resources" cannot be redeclared`);
     }
 
+    if (code.name === 'assign') {
+      for (
```

Comment:
It looks like you want to go into the prototype chain here. Are the values deep inside? In what case you'll have to search the values deeper than the first prototype?

Reply:
If you are in a child scope (eg. inside the curly braces of an if block), your assignment should only affect the variable from your closest parent scope